### PR TITLE
NativeCompilerTarget should support throwing

### PR DESCRIPTION
**Pull request by vchuravy**
As noted in https://github.com/EnzymeAD/Enzyme.jl/pull/243

The issue is that for nested Codegen Enzyme uses the NativeCompilerTarget or PTXCompilerTarget 
and so it picks up the default for each.

### CODE REVIEW

**Comment by vchuravy** [comment]
Hm the windows error I would have thought be fixed with https://github.com/JuliaGPU/GPUCompiler.jl/blob/3eabb5f6d03b42877730fbea9b4a82f8d4721e54/test/definitions/native.jl#L151-L153

**Comment by vchuravy** [comment]
Julia does https://github.com/JuliaLang/julia/blob/811e534ca7398b4856149938995539f77078939d/src/codegen.cpp#L8080-L8084

**Comment by vchuravy** [comment]
Looks good from my side, but this will need a minor bump I think.

**Comment by vchuravy** [comment]
cc: @masonprotter this is probably relevant for you

File: src/native.jl [hunk 1]
```diff
@@ -8,6 +8,7 @@ Base.@kwdef struct NativeCompilerTarget <: AbstractCompilerTarget
     cpu::String=(LLVM.version() < v"8") ? "" : unsafe_string(LLVM.API.LLVMGetHostCPUName())
     features::String=(LLVM.version() < v"8") ? "" : unsafe_string(LLVM.API.LLVMGetHostCPUFeatures())
     always_inline::Bool=false # will mark the job function as always inline
+    jlruntime::Bool=true # Use Julia runtime for throwing errors, instead of the GPUCompiler support
```
Comment (maleadt) [inline src/native.jl:11]: Shouldn't this be used in other places too, then? e.g., not linking the GPUCompiler runtime, or lowering things differently (like the pool alloc intrinsics).

File: Project.toml [hunk 2]
```diff
@@ -14,6 +14,6 @@ UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
 
 [compat]
 ExprTools = "0.1"
-LLVM = "4.7"
+LLVM = "4.8"
```
Comment (maleadt) [inline Project.toml:17]: is this for the ORC stuff?

**Review by maleadt — APPROVED** [review]
LGTM

File: Project.toml (hunk 2 above)
Comment (vchuravy) [reply Project.toml:17]: Yes for `absolute_symbols`

**Pull request merged**